src/guidellm/scheduler/constraints/saturation.py

```diff
@@ -0,0 +1,700 @@
+    :cvar max_window_seconds: Maximum time window for data retention
+    :cvar moe_threshold: Margin of error threshold for slope detection
+    :cvar minimum_ttft: Minimum TTFT threshold for violation counting
+    :cvar maximum_window_ratio: Maximum window size as ratio of total requests
+    :cvar minimum_window_size: Minimum data points required for slope estimation
+    :cvar confidence: Statistical confidence level for t-distribution
+    """
+
+    type_: Literal["over_saturation"] = "over_saturation"  # type: ignore[assignment]
+    enabled: bool = Field(
+        default=True,
+        description="Whether to stop the benchmark if the model is over-saturated",
+    )
+    min_seconds: int | float = Field(
+        default=30.0,
+        ge=0,
+        description="Minimum seconds before checking for over-saturation",
+    )
+    max_window_seconds: int | float = Field(
+        default=120.0,
+        ge=0,
+        description="Maximum over-saturation checking window size in seconds",
+    )
+    moe_threshold: float = Field(
+        default=2.0,
+        ge=0,
+        description="Margin of error threshold for slope detection",
+    )
+    minimum_ttft: float = Field(
+        default=2.5,
+        ge=0,
+        description="Minimum TTFT threshold for violation counting",
+    )
+    maximum_window_ratio: float = Field(
+        default=0.75,
+        ge=0,
+        le=1.0,
+        description="Maximum window size as ratio of total requests",
+    )
+    minimum_window_size: int = Field(
+        default=5,
+        ge=0,
+        description="Minimum data points required for slope estimation",
+    )
+    confidence: float = Field(
+        default=0.95,
+        ge=0,
+        le=1.0,
+        description="Statistical confidence level for t-distribution",
+    )
+
+    def create_constraint(self, **_kwargs) -> Constraint:
+        """
+        Create an OverSaturationConstraint instance from this initializer.
+
+        Constructs a new OverSaturationConstraint with the configuration parameters
+        specified in this initializer. The constraint will be ready for evaluation
+        against scheduler state and requests.
+
+        :param _kwargs: Additional keyword arguments (unused)
+        :return: Configured OverSaturationConstraint instance ready for use
+        """
+        return OverSaturationConstraint(  # type: ignore[return-value]
+            minimum_duration=self.min_seconds,
+            minimum_ttft=self.minimum_ttft,
+            maximum_window_seconds=self.max_window_seconds,
+            moe_threshold=self.moe_threshold,
+            maximum_window_ratio=self.maximum_window_ratio,
+            minimum_window_size=self.minimum_window_size,
+            confidence=self.confidence,
+            enabled=self.enabled,
+        )
+
+    @classmethod
+    def validated_kwargs(
+        cls, over_saturation: bool | dict[str, Any] | None = None, **kwargs
```

Comment:
After [this comment](https://github.com/vllm-project/guidellm/pull/438/files#r2582557671) drop support for `bool` as its handled higher up.


```suggestion
        cls, over_saturation: dict[str, Any] | None = None, **kwargs
```